pages/index.tsx

```diff
@@ -28,22 +28,38 @@ export default function Home() {
   // console.log("*** players", players.data);
   const isMobile = useBreakpointValue({ base: true, md: false });
   const { address } = useWeb3();
-  const hasPlayed = useHasPlayed(address);
-  const hasRegistered = useHasRegistered(address);
+  const playerState = usePlayerState(address);
+  const [heading, setHeading] = useState("");
+  const [action, setAction] = useState(<></>);
+  useEffect(() => {
```

Comment:
2 things you can do here:

1. have a default value for playerState.data (probably with everything false) - so that it's never undefined. something like : `const playerStateData = playerState.data || { hasPlayed: false, hasRegistered: false.... }`

2. if `playerState.isLoading` -> show some sort of loader, then if `playerState.isSuccess`, actually show the header + action area.

3. handle playerData as undefined inside `setHeading` and `<ActionArea`>